downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: ballerina-left-nav-release-notes
+title: 2201.7.0 (Swan Lake) 
+permalink: /downloads/swan-lake-release-notes/2201.7.0/
+active: 2201.7.0
+redirect_from: 
+    - /downloads/swan-lake-release-notes/2201.7.0
+    - /downloads/swan-lake-release-notes/2201.7.0-swan-lake/
+    - /downloads/swan-lake-release-notes/2201.7.0-swan-lake
+    - /downloads/swan-lake-release-notes/
+    - /downloads/swan-lake-release-notes
+---
+
+## Overview of Ballerina Swan Lake 2201.7.0
+
+<em>2201.7.0 (Swan Lake Update 7) is the seventh update release of Ballerina Swan Lake, and it includes a new set of features and significant improvements to the compiler, runtime, standard library, and developer tooling. It is based on the 2023R1 version of the Language Specification.</em> 
+
+## Update Ballerina
+
+Update your current Ballerina installation directly to 2201.7.0 using the [Ballerina Update Tool](/learn/cli-documentation/update-tool/) as follows.
+
+1. Run `bal update` to get the latest version of the Update Tool.
+2. Run `bal dist update` to update to this latest distribution.
+
+## Install Ballerina
+
+If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
+
+## Backward-incompatible changes
+
+- A bug that allowed using a function reference of a `non-isolated` function type in a function or method call expression within an `isolated` function or method has been fixed.
+
+    ```ballerina
+    type Employee record {|
+        string name;
+        int id;
+        string department;
+    |};
+
+    isolated function createEmployee(string[] name,
+            function () returns int idFunction,
+            string department)
+            returns Employee => {
+        name: string:'join(" ", ...name),
+        // Compilation error now, since `idFunction` is not `isolated`.
+        id: idFunction(),
+        department
+    };  
+    ```
+  
+- A bug that caused the broader type to be used instead of the singleton type when a constant is used as a type descriptor has been fixed.
+
+  For example, the code below may result in a compilation error now since `num`, which is of type `CONST` can only be integer `8`.
+
+    ```ballerina
+    const int CONST = 1 + 7;
+    CONST num = 2; // Compilation error now.
+    ```
+
+- A bug that resulted in compilation errors not being logged for duplicate keys via computed name fields of a mapping constructor in a mapping constant declaration has been fixed. This previously resulted in a panic at runtime instead.
+
+    ```ballerina
+    const MESSAGE = "message";
+    const map<string> HTTP_OK = {httpCode: "200", message: "OK", [MESSAGE] : "BAD REQUEST"}; // Compilation error now.
+    ```
+
+- Fixed a bug in the configurable TOML syntax validation for the module structure. The error message thrown for an invalid TOML module structure is now improved to provide the variable name.
```

Comment:
How is this a bug/breaking change?

Reply:
@warunalakshitha @HindujaB @Nadeeshan96

Reply:
Error message changed. So any tests which will depend on previous error message will break.